### PR TITLE
patching CmdSink and adding extensive unit tests

### PR DESCRIPTION
**Pull request by jerrypeng**

### CODE REVIEW

**Comment by sijie** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

File: pulsar-client-tools/pom.xml [hunk 1]
```diff
@@ -86,6 +86,40 @@
...
+        <artifactId>maven-antrun-plugin</artifactId>
+        <executions>
+          <execution>
+            <phase>test-compile</phase>
+            <goals>
+              <goal>run</goal>
+            </goals>
+            <configuration>
+              <tasks>
+                <echo>building python instance</echo>
```
Comment (sijie) [inline pulsar-client-tools/pom.xml]: can you rename to something like "copy test sink package"?

Comment (jerrypeng) [reply pulsar-client-tools/pom.xml]: good catch

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by merlimat** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by sijie** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
rebuild this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Comment by jerrypeng** [comment]
retest this please

**Pull request merged**